### DISCUSSION TITLE
ESP32 C3 SuperMini not supported?

### DISCUSSION THREAD

**Discussion by ChrisVonFrosta** [arendst/Tasmota#20031] [Q&A] [answered]
i have this [board ](https://de.aliexpress.com/item/1005005967641936.html?spm=a2g0o.order_list.order_list_main.244.6b4d5c5fPqlKeA&gatewayAdapt=glo2deu)

When i try to flash it with web installer i got this message:
![grafik](https://github.com/arendst/Tasmota/assets/16272493/6253c244-04b3-4062-b571-dbe53c1d63dc)

what's going on there?

**Comment by sfromis** [comment]
That's just a standard ESP32-C3, and I see the same with another C3. Looking into it, I found that you should be able to flash if you select it under the "development" heading in the drop down menu. Not sure why no C3 bluetooth manifest seems to exist for release, @Jason2866 ?

**Comment by barbudor** [comment]
Just as a side comment, I consider single core CPU including the C3 as not the best for Bluetooth bridge.
I have one, it is working fine until I try to reach the web UI which is then slow.

**Reply by ChrisVonFrosta** [reply]
i didn't knew the difference. Never thought about the core count or that the name ESP32 offers different counts of core/huch performance dif. it is very smal and was cheap.
But it is to slow i have to think about. But as you point on the bluetooth, the non bluetooth image should work more performant?

**Reply by sfromis** [reply]
Yes, without Bluetooth, a Tasmota setup will barely use more than one core. However if you're not using http much (like the web UI), it may still work out with single core, but I'd probably consider if I'd bet on doing additional processing on the chip, like coding to further process Bluetooth data. The ESP32-S2 is also single core, and highest performance is the ESP32-S3.

**Comment by Jason2866** [comment, marked as answer]
There is no official C3 Bluetooth build.

**Reply by sfromis** [reply]
I was expecting development and release to have same builds, with the "other ones" being under "unofficial". Bluetooth under development but not release is also the case for S3.

**Reply by Jason2866** [reply]
Will remove.

**Comment by Staars** [comment]
One more thing:
I just received my SuperMini too and it is indeed super mini.
It uses at the JTAG interface so you need a "C3CDC" build.

**Reply by barbudor** [reply]
I also purchased, received and flashed 2 of them.
But not yet in prod
Indeed very cute ones

I hesitated with the Seedstudio Xiao board but was not keen with the external antenna.
Still wondering the quality of small antenna on that one

**Reply by Staars** [reply]
Good question ... I have pretty uncomplicated WiFi conditions at home, so I am a bad tester.
They were under 2 € in the last few days on sale and I could not resist. In the first few tests they seem to be at least in the "normal" range of snappiness in the web UI. 
Which part of the PCB is supposed to serve as antenna? I can not really spot anything.

**Reply by barbudor** [reply]
The small red thing with "C3" written on it
Things known as ceramic antennas

**Reply by sfromis** [reply]
The antenna is not part of the PCB, but is the small red module at the end, with the "C3" paint on it. A ceramic antenna module having fair chances of being better tuned than PCB antenna designs.

I'm using the quite similar [01Space ESP32-C3FH4-RGB](https://github.com/01Space/ESP32-C3FH4-RGB) (main difference is additional 5x5 neopixels and a Qwiic connector), and have found the small ceramic antenna to work reasonably, except when having too much in wiring adjacent to the antenna, which should not be too surprising anyway.

**Reply by joba-1** [reply]
I also have two of those. They get very warm or even hot quickly. Not sure if it is just this batch...
I then tried the seed studio ones. They work much better, and for me the external antenna is more like a pro.
But true, of course: one core, less MHz is more for the light weight tasks...

**Reply by sfromis** [reply]
I've also found C3 to be warmer than I expected. Like more than luke-warm but not uncomfortably so, running Tasmota with Wifi, without "real" activity.

**Reply by Jason2866** [reply]
Interesting info. I have a few of the cheap luatos CDC C3 boards (PCB antenna). The just get slightly warm and are performing perfectly fine.

**Reply by sfromis** [reply]
Comparing two ESP32-C3 form factors, 01Space and Wemos Mini, I found the latter to be cooler, but attribute that to the heat spreading over a much larger PCB area.

**Reply by brainhackeu** [reply]
I wonder how you got the board working. I try now since a couple of days but i`m stuck getting it hooked up to WiFi. No AP visible and manual IP set / AP config does not work either. Network Scan crashing and rebooting the boards. I used the CDC version, a couple of other from dev to stable branch and tried also to compile myself, no way to get it running WiFi.... BTW: With ESPEasy the same boards are working fine

**Reply by Noschvie** [reply]
@brainhackeu Have you tried to use the Tasmota Web Installer?

**Reply by Jason2866** [reply]
I have such a C3 super mini too. Same behaviour than yours. Had to reduce WiFi Tx power with `WifiPower` to a very small value to be able to connect to WiFi. Seems there are many clones made from.
Mine has horrible wifi performance / antenna.

**Reply by brainhackeu** [reply]
@Noschvie: Yes, i did. i tested almost all variants of C3 (as architecture is different, images for normal ESP32 should not work of course). I used precompiled images downloaded from GIT out of different branches, i tried the Webinstaller (asume this takes the images from related GIT branches as well but i tried it), compiled different branches myself, with CDC, without CDC, no luck. 
Imaging (or compile) works fine, booting also working but once booted : No AP could be found from various WiFi Devices (Android, Windows, Linux, Mobilephone, Tablet, PC, FritzBox ...)  @Jason2866  : Thanks for the hint with TX power, will try that next on the Weekend.

**Reply by Noschvie** [reply]
Oh... seems I have a better one.

**Comment by jkwim** [comment]
I have a ESP32-C3 Super Mini board, similar to OP's.

Does it have a USB controller or the USB is only for power? It does not get detected in Windows

Tried these steps:

Flash Using USB[](https://docs.espressif.com/projects/esp-idf/en/stable/esp32c3/get-started/establish-serial-connection.html#flash-using-usb)

For the ESP32-C3, the USB peripheral is available, allowing you to flash the binaries without the need for an external USB-to-UART bridge.

The USB on the ESP32-C3 uses the GPIO19 for D+ and GPIO18 for D-.

Note

The ESP32-C3 supports only USB CDC and JTAG.

If you are flashing for the first time, you need to get the ESP32-C3 into the download mode manually. To do so, press and hold the BOOT button and then press the RESET button once. After that release the BOOT button.

**Reply by jkwim** [reply]
I even tried installing the driver using Windows Power Shell:

![image](https://github.com/arendst/Tasmota/assets/84637540/b4d6821b-cfe6-4f90-b5b8-965d3b05dbd6)

**Reply by sfromis** [reply]
Tasmota also supports a serial-like connection via USB, if you have flashed a build with USB CDC support, like `tasmota32c3cdc.factory.bin`. Not sure how much you asked about, most of your post is just about how it works, including flashing. And you can use the normal `esptool.py` no problem, however the recommended tool is the [Tasmota Web Installer](https://tasmota.github.io/install/).

**Reply by jkwim** [reply]
Before I can install Tasmota, I need to see a serial port in Windows 11. That's not happening.
So I wonder whether I have to use a USB-TTL adapter for the initial flashing.

**Reply by jkwim** [reply]
PROBLEM SOLVED!
It was a USB cable issue (got mixed up with charging only cable).
